File: backend/src/main/java/com/hazelgym/repository/AttendanceRepository.java
package com.hazelgym.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.hazelgym.model.Attendance;

public interface AttendanceRepository extends JpaRepository<Attendance, Long> {

    List<Attendance> findAllByOrderByFechaHoraDesc();

    List<Attendance> findByUsuarioIdOrderByFechaHoraDesc(Long usuarioId);
}

File: mobile-android/app/src/main/java/com/hazelgym/mobile/data/repository/AttendanceRepository.kt
package com.hazelgym.mobile.data.repository

import com.hazelgym.mobile.data.model.AttendanceCreateRequest
import com.hazelgym.mobile.data.model.AttendanceResponse
import com.hazelgym.mobile.data.model.SessionUser
import com.hazelgym.mobile.data.remote.AttendanceApi

class AttendanceRepository(
    private val attendanceApi: AttendanceApi
) {
    suspend fun getAttendances(sessionUser: SessionUser): List<AttendanceResponse> {
        return attendanceApi.getAttendances("${sessionUser.tokenType} ${sessionUser.token}")
    }

    suspend fun createAttendance(sessionUser: SessionUser, qrCodeId: Long): AttendanceResponse {
        return attendanceApi.createAttendance(
            authorization = "${sessionUser.tokenType} ${sessionUser.token}",
            request = AttendanceCreateRequest(
                usuarioId = sessionUser.userId,
                qrCodeId = qrCodeId
            )
        )
    }
}
